src/main/java/com/kodilla/ecommercee/mapper/CartMapper.java

```diff
@@ -1,17 +1,49 @@
 package com.kodilla.ecommercee.mapper;
 
 import com.kodilla.ecommercee.dto.CartDto;
+import com.kodilla.ecommercee.dto.ItemDto;
 import com.kodilla.ecommercee.entity.Cart;
+import com.kodilla.ecommercee.entity.Item;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
+
 @Component
 public class CartMapper {
 
-    public Cart translateToCart(CartDto cartDto) {
-        return new Cart();
+    @Autowired
+    OrderMapper orderMapper;
+
+    @Autowired
+    ProductMapper productMapper;
+
+
+    public Cart mapToCart(CartDto cartDto) {
+        return new Cart(cartDto.getId(),mapToItemList(cartDto.getItems()),cartDto.isClosed());
+    }
+
+    public CartDto mapToCartDto(Cart cart) {
+        return new CartDto(cart.getId(),mapToItemDtoList(cart.getItems()),cart.isClosed());
     }
 
-    public CartDto translateToCartDto(Cart cart) {
-        return new CartDto();
+    public ItemDto mapToItemDto(Item item){
+        return new ItemDto(mapToCartDto(item.getCart()),productMapper.mapToProductDto(item.getProduct()),item.getQuantity(),item.getPrice());
```

Comment:
Dosyc nie czytelna jest taka dluga linijka. Rozbij to nieco, żeby łatwiej się czytalo